src/io/cuda/cuda_column_data.cpp

```diff
@@ -224,6 +223,7 @@ void CUDAColumnData::ResizeWhenCopySubrow(const data_size_t num_used_indices) {
   #pragma omp parallel for schedule(static) num_threads(num_threads_)
   for (int column_index = 0; column_index < num_columns_; ++column_index) {
     OMP_LOOP_EX_BEGIN();
+    SetCUDADevice(gpu_device_id_, __FILE__, __LINE__);
```

Comment:
do we need to set this inside loop?

Reply:
Move outside. [d4695ff](https://github.com/microsoft/LightGBM/pull/6028/commits/d4695ff37be9cbd2b85de1617252e8e9e883e699)